core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -103,6 +103,11 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.DELETE_TOPICS => handleDeleteTopicsRequest(request)
         case ApiKeys.DELETE_RECORDS => handleDeleteRecordsRequest(request)
         case ApiKeys.INIT_PRODUCER_ID => handleInitPidRequest(request)
+        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleAddPartitionToTransactionRequest(request)
+        case ApiKeys.ADD_OFFSETS_TO_TXN => handleAddOffsetsToTransactionRequest(request)
+        case ApiKeys.END_TXN => handleEndTransactionRequest(request)
+        case ApiKeys.WRITE_TXN_MARKER => handleWriteTxnMarkerRequest(request)
+        case ApiKeys.TXN_OFFSET_COMMIT => handleTxnOffsetCommitRequest(request)
         case requestId => throw new KafkaException("Unknown api code " + requestId)
```

Comment:
Existing issue: we should remove this line so that the compiler can warn us if we have unhandled api keys.